### PR TITLE
1 mini logo

### PR DESCRIPTION
**Pull request by psulliv**

### CODE REVIEW

**Comment by psulliv** [comment]
I hope you like parens:
```
>./1_mini_logo 
./1_mini_logo 
"part b: vector function"
Def "vector" (Pars "x1" (Pars "y1" (Pars "x2" (Name "y2")))) (Seq (Pen Up) (Seq (MoveTo (Names ("x1","y1"))) (Seq (Pen Down) (Seq (MoveTo (Names ("x2","y2"))) (Pen Up)))))
"part c: steps 1"
Seq (Pen Up) (Seq (MoveTo (Ints (1,1))) (Seq (Pen Down) (Seq (MoveTo (Ints (0,1))) (Seq (MoveTo (Ints (0,0))) (Pen Up)))))
"part c: steps 4"
Seq (Pen Up) (Seq (MoveTo (Ints (4,4))) (Seq (Pen Down) (Seq (MoveTo (Ints (3,4))) (Seq (MoveTo (Ints (3,3))) (Seq (Pen Up) (Seq (Pen Up) (Seq (MoveTo (Ints (3,3))) (Seq (Pen Down) (Seq (MoveTo (Ints (2,3))) (Seq (MoveTo (Ints (2,2))) (Seq (Pen Up) (Seq (Pen Up) (Seq (MoveTo (Ints (2,2))) (Seq (Pen Down) (Seq (MoveTo (Ints (1,2))) (Seq (MoveTo (Ints (1,1))) (Seq (Pen Up) (Seq (Pen Up) (Seq (MoveTo (Ints (1,1))) (Seq (Pen Down) (Seq (MoveTo (Ints (0,1))) (Seq (MoveTo (Ints (0,0))) (Pen Up)))))))))))))))))))))))
```

File: hw2/1_mini_logo.hs
Comment (caitlynplewis) [inline hw2/1_mini_logo.hs]: Probably doesn't matter, but the HW problem shows _name_ before _(vals)_ as the parameters for **call**.

File: hw2/1_mini_logo.hs
Comment (caitlynplewis) [inline hw2/1_mini_logo.hs:29]: I appreciate the readability this offers when you use it in the vector definition below. We could add a similar thing for Pars; I'd be happy to do that after this is merged.

File: hw2/1_mini_logo.hs
Comment (caitlynplewis) [inline hw2/1_mini_logo.hs]: It's valid to pattern match on constants, so if we want to, we can get rid of the conditions. I think it's a matter of opinion which you think is better, but I think it reads a little cleaner. 
image

**Review by caitlynplewis — APPROVED** [review]
Looks great! Aside from the readability nightmare that is that many parens...

File: hw2/1_mini_logo.hs
Comment (psulliv) [reply hw2/1_mini_logo.hs]: Oh, it sure is. I'll swap that out

File: hw2/1_mini_logo.hs
Comment (psulliv) [reply hw2/1_mini_logo.hs]: I think your version looks better, I'll switch that up

File: hw2/1_mini_logo.hs
Comment (psulliv) [reply hw2/1_mini_logo.hs:29]: I like that, I'll leave it to you

**Pull request merged**
